.github/workflows/release.yml

```diff
@@ -0,0 +1,59 @@
+name: Release
+
+on:
+  push:
+    tags:
+    - 'v*' # Push events to matching v*, i.e. v1.0, v20.15.10
+
+jobs:
+  release:
+    runs-on: ubuntu-latest
+
+    permissions:
+      id-token: write
+      packages: write
+      contents: write
+
+    env:
+      GOPATH: ${{ github.workspace }}
+      GO111MODULE: on
+      COSIGN_EXPERIMENTAL: "true"
+
+    steps:
+    - uses: actions/setup-go@f6164bd8c8acb4a71fb2791a8b6c4024ff038dab # v2.2.0
+      with:
+        go-version: 1.17.x
+
+    - name: Install ko
+      uses: imjasonh/setup-ko@2c3450ca27f6e6f2b02e72a40f2163c281a1f675 # v0.4
+
+    - name: Install cosign
+      uses: sigstore/cosign-installer@581838fbedd492d2350a9ecd427a95d6de1e5d01 # v2.1.0
+
+    - name: Install GoReleaser
+      uses: goreleaser/goreleaser-action@b953231f81b8dfd023c58e0854a721e35037f28b # v2.9.1
+      with:
+        install-only: true
+
+    - name: Log into ghcr.io
+      uses: docker/login-action@v1
+      with:
+        registry: ghcr.io
+        username: ${{ github.repository_owner }}
+        password: ${{ secrets.GITHUB_TOKEN }}
+
+    - name: Check out code onto GOPATH
+      uses: actions/checkout@a12a3943b4bdde767164f792f33f40b04645d846 # v3.0.0
+      with:
+        fetch-depth: 1
```

Comment:
Nit: I believe this is [the default since v2](https://github.com/actions/checkout/releases/tag/v2.0.0): "Improved fetch performance: The default behavior now fetches only the commit being checked-out"

Reply:
ahh this i forgot to remove